ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -265,16 +265,33 @@ pub(crate) struct Instruction {
     name: RefCell<Option<String>>,
 }
 
-impl Instruction {
+impl<'a> Instruction {
     /// Returns the operand at the specified index. Panics if the index is out of bounds.
     pub(crate) fn operand(&self, index: usize) -> &Operand {
         &self.operands[index]
     }
 
+    /// Return a slice of the remaining operands, starting from the index `from` (inclusive).
+    pub(crate) fn remaining_operands(&self, from: usize) -> &[Operand] {
+        &self.operands[from..]
+    }
+
     pub(crate) fn opcode(&self) -> Opcode {
         self.opcode
     }
 
+    /// For a call instruction, return the callee.
+    ///
+    /// Panics if the instruction isn't a call.
+    pub(crate) fn get_callee(&self, m: &'a Module) -> &'a Function {
```

Comment:
`get_` is not very Rust-like and we should avoid it when possible: `callee` would be fine here I think.

Reply:
Agreed.

Reply:
Fixed in 4b6a16b0d2138270895891c980d0f5f6a1993839.

Other "gets" fixed in:
 - 3e325437f3d12587a91a9aa61ac9346d66f4a18b
 - 3dfe284a77794bbfce6010be7aec635db16d550c

Reply:
I'm fine with the `get`s but I'm not a fan of `Deref` (https://github.com/ykjit/yk/pull/943#discussion_r1466154063) which isn't used for these sorts of purposes - we should have a `to_u16` (or similar) method IMHO to be Rusty. https://doc.rust-lang.org/std/ops/trait.Deref.html's documentation is worth a read:

> Warning: Deref coercion is a powerful language feature which has far-reaching implications for every type that implements Deref. The compiler will silently insert calls to Deref::deref. For this reason, one should be careful about implementing Deref and only do so when deref coercion is desirable. See [below](https://doc.rust-lang.org/std/ops/trait.Deref.html#when-to-implement-deref-or-derefmut) for advice on when this is typically desirable or undesirable.
>
>Types that implement Deref or DerefMut are often called “smart pointers” and the mechanism of deref coercion has been specifically designed to facilitate the pointer-like behaviour that name suggests.

Reply:
Right.

FWIW, I did test the coercion cases I cared about and they didn't seem to work (in a bad way).

Nonetheless, I'm OK with `to_u16()` and/or `to_usize()` depending on use case, so let's do that.

Reply:
I would like to get rid of all the `Deref`s and replace them with explicit methods.

Reply:
I've killed the `Deref`s that were added in this PR in d1ddec800717eeef6e31bd96b533e99a4c8bfcf2.

More pre-existing ones to kill in a future PR (I now have a list of follow-up PRs in my notebook).